### PR TITLE
avahi err handling BIPI-2737 BIPI-2716

### PR DESCRIPTION
**Pull request by simongrac**
- stderr from avahi redirected to stdout, 
- errors are being handled internally, 
- parsing changed due to new avahi-error output: error: vector::_M_range_check should not appear at all
@ondrejsova please can u test its reliability?

### CODE REVIEW

File: include/phoxi_camera/PhoXiInterface.h [hunk 1]
```diff
@@ -88,7 +88,7 @@ namespace phoxi_camera {
         * \param stdoutPipe - avahi-browse call output
         * \return vector of all available scanners on network
         */
-        std::vector<std::string> getAvailableScanersID(const std::vector<std::string> &stdoutPipe);
+        std::vector<std::string> getAvailableScanersID(char type, const std::vector<std::string> &stdoutPipe);
```
[the remark this answers is not in the document]
Comment (simongrac) [reply include/phoxi_camera/PhoXiInterface.h]: clarification added to the describtion of method above, thanks for pointing out

File: src/PhoXiInterface.cpp [hunk 2]
```diff
@@ -388,49 +388,84 @@ namespace phoxi_camera {
...
+        std::vector<std::string> connectedScanners;
+        std::vector<std::string> disappearedScanners;
+        std::vector<std::string> stderrOutput;
+
         if (!pipe) {
+            pclose(pipe);
             throw AvahiFailed("Avahi-browse operation failed");
         }
 
         while (fgets(buffer, sizeof buffer, pipe) != nullptr) {
```
[the remark this answers is not in the document]
Comment (simongrac) [reply src/PhoXiInterface.cpp]: parenthesis added, thanks

File: include/phoxi_camera/PhoXiInterface.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (simongrac) [reply include/phoxi_camera/PhoXiInterface.h]: clarification added to the describtion of method above, thanks for pointing out

[the remark this answers is not in the document]
Comment (simongrac) [reply include/phoxi_camera/PhoXiInterface.h]: changed to enum...

**Comment by najlepsiwebdesigner** [comment]
@simongrac can you please ask @MatejBartosovic to approve this PR, therefore we can proceed with merge?

**Comment by simongrac** [comment]
> @simongrac can you please ask @MatejBartosovic to approve this PR, therefore we can proceed with merge?

Yes, I assigned @ondrejsova for testing and also whote @MatejBartosovic if it is ready for testing (so there is no need of implementation changes) yesterday. I will try to close this asap.

**Comment by ondrejsova** [comment]
I've been using this branch for about a week now, errors didn't appear since.

**Pull request merged**